src/numeric/processes.jl

```diff
@@ -8,36 +10,62 @@ abstract DiscreteProcess <: AbstractProcess
 abstract ContinuousProcess <: AbstractProcess
 abstract IIDExogenous <: AbstractProcess
 
-abstract AbstractDiscretizedProcess <: DiscreteProcess
+abstract AbstractDiscretizedProcess # <: DiscreteProcess
 
+###
+### Discretized process
+###
 
-# discretized process
-
+# date-t grid has a known structure
 type DiscretizedProcess <: AbstractDiscretizedProcess
-    nodes::Matrix{Float64}
-    integration_nodes::Array{Matrix{Float64}}
-    integration_weights::Array{Vector{Float64}}
+    grid::Grid
+    integration_nodes::Array{Matrix{Float64},1}
+    integration_weights::Array{Vector{Float64},1}
 end
 
-n_nodes(dp::DiscretizedProcess) = size(dp.nodes, 1)
+n_nodes(dp::DiscretizedProcess) = n_nodes(dp.grid)
+node(dp::DiscretizedProcess, i) = node(dp.grid,i)
 n_inodes(dp::DiscretizedProcess, i::Int) = size(dp.integration_nodes[i], 1)
-inodes(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_nodes[i][j, :]
-iweights(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_weights[i][j]
-node(dp::DiscretizedProcess, i) = dp.nodes[i, :]
+inode(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_nodes[i][j, :]
+iweight(dp::DiscretizedProcess, i::Int, j::Int) = dp.integration_weights[i][j]
+
+# date-t grid is unstructured
+type DiscreteMarkovProcess <: AbstractDiscretizedProcess
+    grid::UnstructuredGrid
+    transitions::Matrix{Float64}
+    values::Matrix{Float64}
+end
+
+DiscreteMarkovProcess(transitions::Matrix{Float64}, values::Matrix{Float64}) =
```

Comment:
Good catch, that's a typo. In the case of Markov chains, the nodes attainable today and tomorrow are the same, and they are given by `values`